src/debugging/DockerServerReadyAction.ts

```diff
@@ -213,54 +190,187 @@ export class ServerReadyDetector extends vscode.Disposable {
     }
 }
 
-export function registerServerReadyAction(context: vscode.ExtensionContext): void {
+type DebugAdapterMessage = {
+    body?: {
+        category?: string;
+        output?: string;
+    };
+    type?: string;
+    event?: string;
+};
+
+interface DockerServerReadyManager {
```

Comment:
I kind of want to rename this something like `DockerServerReadyDetector`; "manager" doesn't really help me understand the difference between the two implementations.